### ISSUE TITLE
Add `xml` module to parse and emit XML

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#835]
We need some way to parse XML data... so let's write a module...

Does it make sense to use libxml2 or some such? I really don't like this kind of advanced parsing in C (and the security bugs that will invariable come as a consequence) but we have to move forward..

- [ ] add libxml2 to system @plajjan 
- [ ] write xml module @sydow

**Comment by plajjan** [comment]
When decoding an XML string, convert into a tree of Tag objects, something like this:
```
class Tag():
    def __init__(self, name: str, namespace: str, attributes: dict[str, str], children: list[Tag]):
        pass

r = xml.Decode("""<hej apa="bepa"><foo/></hej>""")

r = Tag("hej", None,
        {"apa": "bepa"},
        [
            Tag("foo", None, {}, [])
        ]
        )


```

I think we can assume that the input is utf-8 (it will be for the first use cases)...

**Comment by sydow** [comment]
To avoid misunderstandings: the branch xmlparser has a first implementation of this, but I do not try to merge this, since it requires libraries l -lz -llzma -liconv -lxml2 to be added to the cc command and I wait for a strategy to require this.

**Comment by plajjan** [comment]
@sydow I've been meaning to help out with that. FWIW, the short term solution is to just add -l args. Longer term my idea was to bake all object files into one big archive which we can then specify and it's up to the linker to pick what it wants from there, see #849.

**Comment by plajjan** [comment]
Fixed with #990. Closing.
